### PR TITLE
Compat update for String rename

### PR DESCRIPTION
**Pull request by StefanKarpinski**

### CODE REVIEW

File: REQUIRE [hunk 1]
```diff
@@ -2,3 +2,4 @@ julia 0.4
 JSON
 Blink 0.3.3
 Colors
+Compat
```
Comment (yuyichao) [inline REQUIRE]: `0.7.15`

File: src/traces_layouts.jl [hunk 2]
```diff
@@ -2,7 +2,7 @@ abstract AbstractTrace
 abstract AbstractLayout
 
 type GenericTrace{T<:Associative{Symbol,Any}} <: AbstractTrace
-    kind::ASCIIString
+    kind::String
```
Comment (yuyichao) [inline src/traces_layouts.jl]: This looks reasonably private.

Comment (StefanKarpinski) [inline src/traces_layouts.jl]: ??

Comment (yuyichao) [inline src/traces_layouts.jl]: This is a breaking change but it looks reasonably private so it should be fine it it passes CI.

**Comment by spencerlyon2** [comment]
Nice! So happy this landed in base.

The CI failures are the usual issues with the web stack not building on Linux. 

The OS X tests pass, which makes me think this works without a problem, conditional on dependencies getting installed.

**Pull request merged**